services/common/src/redux/slices/permitSearchSlice.ts

```diff
@@ -0,0 +1,109 @@
+import { createAppSlice } from "@mds/common/redux/createAppSlice";
+import { hideLoading, showLoading } from "react-redux-loading-bar";
+import CustomAxios from "@mds/common/redux/customAxios";
+import { ENVIRONMENT } from "@mds/common/constants/environment";
+import { RootState } from "../rootState";
+import { createRequestHeader } from "../utils/RequestHeaders";
+import { Facet, SearchQuery, SearchResult } from "@mds/common/interfaces/search/facet-search.interface";
+
+export const permitSearchReducerType = "permitSearch";
+
+interface PermitSearchState {
+    results: SearchResult | null;
+    loading: boolean;
+    query: string;
+    filters: Array<{ category: string; value: string }>;
+    allFacets: { [key: string]: Facet[] };
+}
+
+const initialState: PermitSearchState = {
+    results: null,
+    loading: false,
+    query: '',
+    filters: [],
+    allFacets: {},
+};
+
+const permitSearchSlice = createAppSlice({
+    name: permitSearchReducerType,
+    initialState,
+    reducers: (create) => ({
+        setQuery: create.reducer((state, action: { payload: string }) => {
+            state.query = action.payload;
+        }),
+        setFilters: create.reducer((state, action: { payload: Array<{ category: string; value: string }> }) => {
+            state.filters = action.payload;
+        }),
+        searchPermitConditions: create.asyncThunk(
+            async (payload: SearchQuery, thunkApi) => {
+                thunkApi.dispatch(showLoading());
+                const headers = createRequestHeader();
+
+                try {
+                    const response = await CustomAxios().post(
+                        `${ENVIRONMENT.apiUrl}/search/permit-conditions`,
+                        payload,
+                        headers
+                    );
+
+                    thunkApi.dispatch(hideLoading());
+                    return response.data;
+                } catch (error) {
+                    thunkApi.dispatch(hideLoading());
+                    throw error;
```

Comment:
is it expected that there's an error creating the post call? Otherwise I'm not sure that there's a purpose to catching and throwing the error